src/promptflow/promptflow/_sdk/_submitter/test_submitter.py

```diff
@@ -412,31 +415,76 @@ def flow_test(
                 )
 
                 line_result: LineResult = async_run_allowing_running_loop(
-                    flow_executor.exec_line_async, inputs, index=0
+                    flow_executor.exec_line_async, inputs, index=0, allow_generator_output=allow_generator_output
                 )
-                line_result.output = persist_multimedia_data(
-                    line_result.output, base_dir=self.flow.code, sub_dir=Path(".promptflow/output")
-                )
-                if line_result.aggregation_inputs:
-                    # Convert inputs of aggregation to list type
-                    flow_inputs = {k: [v] for k, v in inputs.items()}
-                    aggregation_inputs = {k: [v] for k, v in line_result.aggregation_inputs.items()}
-                    aggregation_results = async_run_allowing_running_loop(
-                        flow_executor.exec_aggregation_async, flow_inputs, aggregation_inputs
+                if isinstance(line_result, AsyncGeneratorType):
```

Comment:
This function is long, may refactor to get it shorter

Reply:
updated